companion-select-text-insert/selected-text-below.qml

```diff
@@ -0,0 +1,103 @@
+import QtQml 2.0
+import QOwnNotesTypes 1.0
+
+Script {
+    /**
+     * Writes the selected text on the webpage that was sent to QOwnNotes from
+     * the browser extension into the current note
+     */
+ 
+    property string linkStyle;
+
+    // register your settings variables so the user can set them in the script settings
+    //
+    // unfortunately there is no QVariantHash in Qt, we only can use
+    // QVariantMap (that has no arbitrary ordering) or QVariantList (which at
+    // least can be ordered arbitrarily)
+    property variant settingsVariables: [
+        {
+            "identifier": "linkStyle",
+            "name": "Insert Link as Ref Style",
+            "description": "Insert link As Reference , Below or NoLink ?",
+            "type": "selection",
+            "default": "Below",
+            "items": {"Below": "Below", "Ref": "Ref", "NoLink": "NoLink"},
+        }
+    ]
+    function websocketRawDataHook(requestType, pageUrl, pageTitle, rawData,
+                                  screenshotDataUrl) {
+        // we only want to handle selection requests, page requests are ignored
+        if (requestType != "selection") {
+            return false;
+        };
+
+        switch (linkStyle) {
+            case "Below":
+                script.log(linkStyle)
```

Comment:
Do you need all those script.log()?

Reply:
I will remove now , added for debugging because the options are not selecting , turn out that the values are keys not values.